test/common/helpers.test.js

```diff
@@ -150,3 +150,72 @@ test('dataObjectUriToHttps should throw a Error when passed an invalid uri', (t)
 test('samBaseUrl should come from the config json', (t) => {
     t.is(samBaseUrl(), config.samBaseUrl);
 });
+
+/**
+ * Test the getHashesMap() function
+ */
+
+test('getHashesMap should return empty object for empty checksums array', (t) => {
+    t.deepEqual(getHashesMap([]), {});
+});
+
+test('getHashesMap should return map with 1 entry for checksums array with 1 element', (t) => {
+    const checksumArray = [
+        {
+            type: 'md5',
+            checksum: '336ea55913bc261b72875bd259753046'
+        }
+    ];
+    const expectedChecksumMap = {
+        md5: '336ea55913bc261b72875bd259753046'
+    };
+
+    t.deepEqual(getHashesMap(checksumArray), expectedChecksumMap);
+});
+
+test('getHashesMap should return map with multiple hashes for checksums array', (t) => {
+    const checksumArray = [
+        {
+            type: 'md5',
+            checksum: '336ea55913bc261b72875bd259753046'
+        },
+        {
+            type: 'sha256',
+            checksum: 'f76877f8e86ec3932fd2ae04239fbabb8c90199dab0019ae55fa42b31c314c44'
+        },
+        {
+            type: 'crc32c',
+            checksum: '8a366443'
+        },
+    ];
+    const expectedChecksumMap = {
+        md5: '336ea55913bc261b72875bd259753046',
+        sha256: 'f76877f8e86ec3932fd2ae04239fbabb8c90199dab0019ae55fa42b31c314c44',
+        crc32c: '8a366443'
+    };
+
+    t.deepEqual(getHashesMap(checksumArray), expectedChecksumMap);
+});
+
+test('getHashesMap should return map for checksums array containing multiple `checksum` for same `type`', (t) => {
```

Comment:
[TOL] I would mildly prefer that we halt and report an error in this case because repeated checksums seems like illegal data

Reply:
@kshakir What would be your opinion on this?

Reply:
How about-- @salonishah11 if you can complete the change plus update the test in less than 1hr, I say go for @aednichols' suggestion. If you get to 1 hour and you haven't finished, leave it as is?

There absolutely should never be two of the same checksum types returned... so this check should not be required. But there are so many edge cases in the various DOS/DRS versions and implementations that we keep bumping into...

In this case, with the halt-and-report, we'll receive early notification of the issue versus something strange happening for the user and one of us having to track down the mystery.

Reply:
I have updated the code to throw an error if it encounters a duplicate checksum value for a hash type